Assets/Tests/InputSystem/Plugins/PlayerInputTests.cs

```diff
@@ -826,6 +829,25 @@ public void PlayerInput_ByDefaultChoosesMostSpecificControlSchemeAvailable()
         Assert.That(playerInput.currentControlScheme, Is.EqualTo("PS4Gamepad"));
     }
 
+    // https://fogbugz.unity3d.com/f/cases/1214519/
+    [Test]
+    [Category("PlayerInput")]
+    public void PlayerInput_CanHaveSpacesAndSpecialCharactersInActionNames()
+    {
+        InputSystem.AddDevice<Gamepad>();
+
+        var go = new GameObject();
+        var playerInput = go.AddComponent<PlayerInput>();
+        playerInput.defaultActionMap = "gameplay";
+        playerInput.notificationBehavior = PlayerNotifications.SendMessages;
+        playerInput.actions = InputActionAsset.FromJson(kActions);
+        var listener = go.AddComponent<MessageListener>();
+
+        Press((ButtonControl)playerInput.actions["Action With Spaces!!"].controls[0]);
```

Comment:
Maybe we could have a parametric test with a few different variations on special characters and spaces

Reply:
Probably. Opting on the lazy side for this one :)